File: Count Subarrays With Fixed Bounds.js
/**
 * @param {number[]} nums
 * @param {number} minK
 * @param {number} maxK
 * @return {number}
 */
var countSubarrays = function(nums, minK, maxK) 
{
    const n = nums.length;
    let left = -1;
    let maxi = -1, mini = -1;
    let count = 0;

    for(let i = 0; i < n; i++)
    {
        if(nums[i] < minK || nums[i] > maxK)
        {
            left = i;
        }
        if(nums[i] == minK)
        {
            mini = i;
        }
        if(nums[i] == maxK)
        {
            maxi = i;
        }
        count += Math.max(0 , Math.min(maxi, mini) - left);
    }
    return count;
};

File: Count Subarrays With Fixed Bounds.py
class Solution(object):
    def countSubarrays(self, nums, minK, maxK):
        
        count = 0
        left = mini = maxi = -1

        for i in range(len(nums)):
            if nums[i] < minK or nums[i] > maxK:
                left = i
            
            if nums[i] == minK:
                mini = i

            if nums[i] == maxK:
                maxi = i

            count += max(0, min(maxi, mini) - left)
        
        return count
        
        